### PR TITLE
Code Refactoring and Re-Organization

### PR DESCRIPTION
**Pull request by renatonascalves**
There are several changes here and sorry for going a bit overboard.

Mainly I'm suggesting a "better" file/folder organization and trying to bring some code changes to improve documentation, reading, and understanding of the codebase.

I'm following some "best" practices, mainly from the WooCommerce/Gutenberg projects.

Hopefully, those changes will set as a good base for future blocks and others following our footsteps. :)

Let me know what you think.

### CODE REVIEW

**Review by imath — CHANGES_REQUESTED** [review]
Hi @renatonascalves 

Thanks a lot for your work on this. Unfortunately I can't approve it. I agree code organization is really nice, but we need to make sure there are no feature regressions. There's at least one I've found that made me stopped my tests of your PR:

When publishing an activity, the Publish button should animate itself, the content of editor should be cleared and a notice should allow me to view the just posted activity.

See this video for the expected behavior: [video](https://cldup.com/tN_FKQaoD5.mp4)

I would advise you to test the Editor, the block to share a post into the activity stream and the one to embed an activity using the master branch to be sure your edits do not generate regressions about the features 😉 

Also, JavaScript indentation style is wrong. As [WordPress JavaScript Coding standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/) requires: `tab` must be used instead of 2 spaces.

**Comment by renatonascalves** [comment]
@imath Awesome! I was expecting more push back from those changes, but I'm happy when you are open to that.

I'm going to address those issues and request another review when ready. =D

File: build/bp-activity/bp-activity-blocks.php [hunk 1]
```diff
@@ -380,7 +380,7 @@ function bp_activity_render_share_activity_block( $attributes = array() ) {
 		array(
 			'text'                => '',
 			'borderRadius'        => '',
-			'style'               => '',
+			'style'               => array(),
```
Comment (renatonascalves) [inline build/bp-activity/bp-activity-blocks.php:383]: I changed this into an `array` since this is what's expected and it was causing `undefined` errors.

**Comment by renatonascalves** [comment]
@imath Updated to account for:

- the indentation issues: we are using tabs now
- the correct actions are being fired when saving the activity

I tested the blocks and the activity editor and they seem to be working as expected now.

:)

**Review by imath — APPROVED** [review]
I confirm it fixed the regression. Thanks a lot for your great work on this PR @renatonascalves. I agree your reorganization is really nice. Let's have it in!

**Pull request merged**